src/TLuaInterpreter.cpp

```diff
@@ -12029,6 +12029,41 @@ bool TLuaInterpreter::compile(const QString & code, QString & errorMsg )
     }
 
     int error = luaL_dostring( L, code.toLatin1().data() );
+
+    QString n;
+    if( error != 0 )
+    {
+        string e = "Lua syntax error:";
+        if( lua_isstring( L, 1 ) )
+        {
+            e.append( lua_tostring( L, 1 ) );
+        }
+        errorMsg = "<b><font color='blue'>";
+        errorMsg.append( e.c_str() );
+        errorMsg.append("</b></font>");
+        if( mudlet::debugMode ){ TDebug(QColor(Qt::white),QColor(Qt::red))<<"\n "<<e.c_str()<<"\n">>0;}
+    }
+    else
+    {
+        if( mudlet::debugMode ){ TDebug(QColor(Qt::white),QColor(Qt::darkGreen))<<"\nLUA: code compiled without errors. OK\n" >> 0;}
+    }
+    lua_pop( L, lua_gettop( L ) );
+
+    if( error == 0 ) return true;
+    else return false;
+}
+
+bool TLuaInterpreter::compile(const QString & code, QString & errorMsg, const QString & name)
```

Comment:
Hrm. So I was able to replace the 1-arg and 2-arg versions without issue.

I did, however, find an interesting bug. Instead of passing `getName()` as the name of the object, I used things like `getName().prepend("Alias: ")` so that the message was as clear as possible. Apparently though, TTimers are the only of the object types whose `getName()` method returns a reference rather than just a string. This led to prepend actually changing the name of the script, and it was hilarious to see:

![http://puu.sh/tE5SZ/02b522360c.png](http://puu.sh/tE5SZ/02b522360c.png)